cmd/query/app/http_handler.go

```diff
@@ -466,11 +487,20 @@ func (aH *APIHandler) archiveTrace(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// QueryService.ArchiveTrace can now archive this traceID.
-	// TODO: add start time & end time
+	startTime, err := aH.queryParser.parseTime(r, startTimeParam, time.Microsecond)
+	if aH.handleError(w, err, http.StatusBadRequest) {
+		return
+	}
+	endTime, err := aH.queryParser.parseTime(r, endTimeParam, time.Microsecond)
+	if aH.handleError(w, err, http.StatusBadRequest) {
+		return
+	}
 	query := spanstore.GetTraceParameters{
-		TraceID: traceID,
+		TraceID:   traceID,
+		StartTime: startTime,
+		EndTime:   endTime,
```

Comment:
from the start of the function up to this point it's the same code as in getTrace. Please move it to `parseGetTraceParameters` helper

Reply:
done